### PR TITLE
paramButtons: respect lockZeroParamFunctions and only create buttons when minArgs, maxArgs allow it

### PR DESCRIPTION
**Pull request by cpirich**
* When `lockZeroParamFunctions` is `true`, we don't show the `addButton` or `subtractButton` for function calls that currently have zero parameters
* Selectively show the `addButton` and `subtractButton` in function calls based on `minArgs` and `maxArgs` data that may have been supplied as part of the list of `known` functions. Now, you won't see a `subtractButton` when you are already using the minimum number of arguments, and similarly, you won't see an `addButton` when you are already using the maximum number of arguments.

### CODE REVIEW

**Comment by cpirich** [comment]
@Bjvanminnen

File: src/languages/javascript.coffee [hunk 1]
```diff
@@ -593,14 +593,22 @@ exports.JavaScriptParser = class JavaScriptParser extends parser.Parser
       when 'CallExpression', 'NewExpression'
         known = @lookupKnownName node
         blockOpts = {}
+        argCount = node.arguments.length
         if known?.fn
           showButtons = known.fn.minArgs? or known.fn.maxArgs?
+          minArgs = known.fn.minArgs or 0
+          maxArgs = if known.fn.maxArgs? then known.fn.maxArgs else Infinity
```
Comment (Bjvanminnen) [inline src/languages/javascript.coffee]: I think coffeescript allows us to write this more tersely as `known.fn.maxArgs ? Infinity`

**Comment by Bjvanminnen** [comment]
one nit, otherwise lgtm

File: src/languages/javascript.coffee (hunk 1 above)
Comment (cpirich) [inline src/languages/javascript.coffee]: Nice, I was trying to figure that out. Will correct in next PR, coming your way shortly..

**Pull request merged**